src/Components/ImagesTable/ClonesTable.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from 'react';
+
+import { ClipboardCopy, Skeleton } from '@patternfly/react-core';
+import {
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+
+import { StatusClone, AwsDetailsStatus } from './Status';
+
+import {
+  ClonesResponseItem,
+  ComposesResponseItem,
+  UploadStatus,
+  useGetCloneStatusQuery,
+  useGetComposeStatusQuery,
+} from '../../store/imageBuilderApi';
+
+type RowPropTypes = {
+  ami: JSX.Element;
+  region: JSX.Element;
+  status: JSX.Element;
+};
+
+type AmiPropTypes = {
+  status: UploadStatus | undefined;
+};
+
+const Ami = ({ status }: AmiPropTypes) => {
+  return status?.status === 'success' ? (
+    <ClipboardCopy hoverTip="Copy" clickTip="Copied" variant="inline-compact">
+      {'ami' in status.options ? status.options.ami : null}
+    </ClipboardCopy>
+  ) : (
+    <Skeleton width="12rem" />
+  );
+};
+
+const ComposeRegion = () => {
+  return <p>us-east-1</p>;
+};
+
+type CloneRegionPropTypes = {
+  region: string;
+};
+
+const CloneRegion = ({ region }: CloneRegionPropTypes) => {
+  return <p>{region}</p>;
+};
+
+const Row = ({ ami, region, status }: RowPropTypes) => {
+  return (
+    <Tbody>
+      <Tr className="no-bottom-border">
+        <Td dataLabel="AMI">{ami}</Td>
+        <Td dataLabel="Region">{region}</Td>
+        <Td dataLabel="Status">{status}</Td>
+      </Tr>
+    </Tbody>
+  );
+};
+
+type CloneRowPropTypes = {
+  clone: ClonesResponseItem;
+};
+
+const CloneRow = ({ clone }: CloneRowPropTypes) => {
+  const [pollingInterval, setPollingInterval] = useState(8000);
+
+  const { data: status, isSuccess } = useGetCloneStatusQuery(
+    {
+      id: clone.id,
+    },
+    { pollingInterval: pollingInterval }
+  );
+
+  useEffect(() => {
+    if (status?.status === 'success' || status?.status === 'failure') {
+      setPollingInterval(0);
+    } else {
+      setPollingInterval(8000);
+    }
+  }, [setPollingInterval, status]);
+
+  switch (true) {
+    case isSuccess:
+      return (
+        <Row
+          ami={<Ami status={status} />}
+          region={<CloneRegion region={clone.request.region} />}
+          status={<StatusClone clone={clone} status={status} />}
+        />
+      );
+    default:
+      return <LoadingRow clone={clone} />;
+  }
+};
```

Comment:
This is a standard pattern to replace if/else chains in Javascript, I'm using it here because I find it improves readability. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#an_alternative_to_if...else_chains

In this case, if isSuccess === true, we render the `<Row>`, otherwise we render `<LoadingRow>`.

Reply:
Alright, I didn't know it was a standard pattern, I'm not used to it so it makes me not liking it, but that's just syntactic sugar